eng/download-source-built-archive.sh

```diff
@@ -72,16 +80,31 @@ function DownloadArchive {
   local archiveUrl
   local artifactsBaseFileName="Private.SourceBuilt.Artifacts"
   local prebuiltsBaseFileName="Private.SourceBuilt.Prebuilts"
+  local sdkBaseFileName="dotnet-sdk-"
   local defaultArtifactsRid='centos.10-x64'
 
-  if [[ "$propertyName" == "MicrosoftNETSdkVersion" ]]; then
-    archiveUrl="https://ci.dot.net/public/source-build/$artifactsBaseFileName.$archiveVersion.$artifactsRid.tar.gz"
-  elif [[ "$propertyName" == *Prebuilts* ]]; then
-    archiveUrl="https://builds.dotnet.microsoft.com/source-built-artifacts/assets/$prebuiltsBaseFileName.$archiveVersion.$defaultArtifactsRid.tar.gz"
-  elif [[ "$propertyName" == *Artifacts* ]]; then
-    archiveUrl="https://builds.dotnet.microsoft.com/source-built-artifacts/assets/$artifactsBaseFileName.$archiveVersion.$artifactsRid.tar.gz"
-  elif [[ "$propertyName" == *Sdk* ]]; then
-    archiveUrl="https://builds.dotnet.microsoft.com/source-built-artifacts/sdks/dotnet-sdk-$archiveVersion-$artifactsRid.tar.gz"
+  # Use override base filename if provided
+  if [[ -n "$archiveBaseFileNameOverride" ]]; then
+    artifactsBaseFileName="$archiveBaseFileNameOverride"
+    prebuiltsBaseFileName="$archiveBaseFileNameOverride"
+    sdkBaseFileName="$archiveBaseFileNameOverride"
+  fi
+
+  local versionDelimiter="."
+  if [[ "$propertyName" == "PrivateSourceBuiltSdkVersion" ]] || [[ "$archiveBaseFileNameOverride" == *sdk* ]]; then
+    versionDelimiter="-"
+  fi
+
+  archiveVersion="${versionDelimiter}${archiveVersion}${versionDelimiter}"
+
+  if [[ "$propertyName" == "$sdkVersionProperty" ]]; then
+    archiveUrl="https://ci.dot.net/public/source-build/${artifactsBaseFileName}${archiveVersion}${artifactsRid}.tar.gz"
```

Comment:
- On release day, do we move the internal source-built archives to the public CI storage?
- The SDK version prop will be non-stable always. WIll that match up with the archive version?

Reply:
> On release day, do we move the internal source-built archives to the public CI storage?

We will need to. I logged https://github.com/dotnet/release/issues/1575

> The SDK version prop will be non-stable always. WIll that match up with the archive version?

Yes, we'll ensure that we always publish with non-stable version.